code/modules/mining/fulton.dm

```diff
@@ -0,0 +1,168 @@
+			balloon2.pixel_y = 10
+			balloon2.appearance_flags = RESET_COLOR | RESET_ALPHA | RESET_TRANSFORM
+			holder_obj.overlays += balloon2
+			sleep(4)
+			balloon = image('icons/obj/fulton_balloon.dmi',"fulton_balloon")
+			balloon.pixel_y = 10
+			balloon.appearance_flags = RESET_COLOR | RESET_ALPHA | RESET_TRANSFORM
+			holder_obj.overlays -= balloon2
+			holder_obj.overlays += balloon
+			playsound(holder_obj.loc, 'sound/items/fulext_deploy.wav', 50, 1, -3)
+			animate(holder_obj, pixel_z = 10, time = 20)
+			sleep(20)
+			animate(holder_obj, pixel_z = 15, time = 10)
+			sleep(10)
+			animate(holder_obj, pixel_z = 10, time = 10)
+			sleep(10)
+			animate(holder_obj, pixel_z = 15, time = 10)
+			sleep(10)
+			animate(holder_obj, pixel_z = 10, time = 10)
+			sleep(10)
+			playsound(holder_obj.loc, 'sound/items/fultext_launch.wav', 50, 1, -3)
+			animate(holder_obj, pixel_z = 1000, time = 30)
+			if(istype(A, /mob/living/carbon/human))
+				var/mob/living/carbon/human/L = A
+			//	L.forcesay(extraction_appends) // ";HELP BL-AAAAAAAAAAAAAAAAHHHHHHHHH"
+				L.SetParalysis(0) // wakey wakey
+				L.drowsyness = 0
+				L.sleeping = 0
+			sleep(30)
+			var/list/flooring_near_beacon = list()
+			for(var/turf/open/floor in orange(1, beacon))
+				flooring_near_beacon += floor
+			holder_obj.loc = pick(flooring_near_beacon)
+			animate(holder_obj, pixel_z = 10, time = 50)
+			sleep(50)
+			animate(holder_obj, pixel_z = 15, time = 10)
+			sleep(10)
+			animate(holder_obj, pixel_z = 10, time = 10)
+			sleep(10)
+			balloon3 = image('icons/obj/fulton_balloon.dmi',"fulton_retract")
+			balloon3.pixel_y = 10
+			balloon3.appearance_flags = RESET_COLOR | RESET_ALPHA | RESET_TRANSFORM
+			holder_obj.overlays -= balloon
+			holder_obj.overlays += balloon3
+			sleep(4)
+			holder_obj.overlays -= balloon3
+			A.anchored = 0 // An item has to be unanchored to be extracted in the first place.
+			A.density = initial(A.density)
+			animate(holder_obj, pixel_z = 0, time = 5)
+			sleep(5)
+			A.loc = holder_obj.loc
+			qdel(holder_obj)
+			if(uses_left <= 0)
+				qdel(src)
+
+/obj/machinery/extraction_point
+	name = "Fulton Recovery Beacon"
+	desc = "A beacon for the fulton recovery system. Hit a beacon with a pack to link the pack to a beacon."
+	icon = 'icons/obj/fulton.dmi'
+	icon_state = "extraction_point"
+	anchored = 1
+	density = 0
+	var/beacon_network = "station"
+
+/obj/machinery/extraction_point/New()
+	var/area/area_name = get_area(src)
+	name += " ([rand(100,999)]) ([area_name.name])"
+	total_extraction_beacons += src
+	..()
+
+/obj/machinery/extraction_point/Destroy()
+	total_extraction_beacons -= src
+	..()
+
+/obj/effect/extraction_holder
+	name = "extraction holder"
+	desc = "you shouldnt see this"
+	var/atom/movable/stored_obj
+
+
+
+/obj/item/weapon/extraction_pack/proc/check_for_living_mobs(atom/A)
+	if(istype(A, /mob/living))
+		var/mob/living/L = A
+		if(L.stat != DEAD)
+			return 1
+	for(var/thing in A)
+		if(.(thing))
```

Comment:
I guess I could just do the contents. Is there anything you can put a mob 2 layers deep into?